plugin/src/Reconciler/applyPatch.lua

```diff
@@ -88,10 +88,10 @@ local function applyPatch(instanceMap, patch)
 		}
 		local partiallyApplied = false
 
-		-- If the instance's className changed, we have a bumpy ride ahead while
-		-- we recreate this instance and move all of its children into the new
-		-- version atomically...ish.
-		if update.changedClassName ~= nil then
+		-- If the instance's className changed, or there is custom recreate logic,
+		-- we have a bumpy ride ahead while we recreate this instance and move all 
+		-- of its children into the new version atomically...ish.
+		if update.requiresRecreate then
```

Comment:
Instead of having `requiresRecreate` as a property here, we should check the conditions that would cause us to recreate an instance directly. For now, that's just `changedClassName ~= nil or changedProperties["MeshId"] ~= nil`.